File: C#/prog_lab_6/point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace prog_lab_6
{
    internal class point
    {
        private double x, y;
        private string discription = "Empty";

        public double X//свойства
        {
            get { return x; }
            set { x = value; }
        }
        public double Y
        {
                get { return y; }
                set { y = value; }
        }
        public string Discription {
            get { return discription; }
            set { discription = value; }
        }
        public void init(double x, double y)
        {//Использование this
            this.x = x;
            this.y = y;
        }
        public static point operator +(point a, point b)//перегрузка +
        {
            point c = new point();
            c.x = a.x + b.x;
            c.y = a.y + b.y;
            return c;
        }
        public static point operator ++(point a)//перегрузка ++
        {
            point b = new point();
            b.x = a.x + 1;
            b.y = a.y + 1;
            return b;
        }
        public void vvod()
        {
            while (!double.TryParse(Console.ReadLine(), out x))
                Console.WriteLine("Координата x введена неверно.");
            while (!double.TryParse(Console.ReadLine(), out y))
                Console.WriteLine("Координата y введена неверно.");
        }
        public void vivod()
        {
            Console.Write($"x={x}, y={y}");
        }
        public double distance(point sdpoint)
        {
            return Math.Sqrt(Math.Pow(sdpoint.x - x, 2) + Math.Pow(sdpoint.y - y, 2));
        }
        public double getx()
        {
            return x;
        }
        public double gety()
        {
            return y;
        }
    }
}

File: Java/prog_lab_7/src/prog_lab_7/point.java
package prog_lab_7;
import java.util.Scanner;

public class point {
	double x, y;
	String description;
	
	public point() {
		description = new String();
	}
	
	public void vvod() {
		Scanner in = new Scanner(System.in);
		this.x = in.nextDouble();
		this.y = in.nextDouble();
	}
	
	public void vivod()	{
		System.out.printf("(%f; %f)\n", this.x, this.y);
	}

	public double distance(point sdpoint) {
		return Math.sqrt(Math.pow(this.x - sdpoint.x, 2) + Math.pow(this.y - sdpoint.y, 2));
	}
	
	public double getx() {
		return this.x;
	}
	
	public double gety() {
		return this.y;
	}
	
	public void init(double x, double y) {
		this.x = x;//использование this
		this.y = y;
		return;
	}
	
	public void descrin() {
		Scanner in = new Scanner(System.in);
		description = in.nextLine();
	}
	
	public void descrout() {
		System.out.println(description);
	}
	
	public int sum() {
		return (int)(x+y);
	}
}